tools/Microsoft.DotNet.XHarness.SimulatorInstaller/Commands/SimulatorInstallerCommand.cs

```diff
@@ -0,0 +1,238 @@
+            {
+                if (downloadable == null)
+                {
+                    continue;
+                }
+
+                var nameNode = downloadable.SelectSingleNode("key[text()='name']/following-sibling::string");
+                var versionNode = downloadable.SelectSingleNode("key[text()='version']/following-sibling::string");
+                var sourceNode = downloadable.SelectSingleNode("key[text()='source']/following-sibling::string");
+                var identifierNode = downloadable.SelectSingleNode("key[text()='identifier']/following-sibling::string");
+                var fileSizeNode = downloadable.SelectSingleNode("key[text()='fileSize']/following-sibling::integer|key[text()='fileSize']/following-sibling::real");
+                var installPrefixNode = downloadable.SelectSingleNode("key[text()='userInfo']/following-sibling::dict/key[text()='InstallPrefix']/following-sibling::string");
+
+                var version = versionNode.InnerText;
+                var versions = version.Split('.');
+                var versionMajor = versions[0];
+                var versionMinor = versions[1];
+                var dict = new Dictionary<string, string>() {
+                    { "DOWNLOADABLE_VERSION_MAJOR", versionMajor },
+                    { "DOWNLOADABLE_VERSION_MINOR", versionMinor },
+                    { "DOWNLOADABLE_VERSION", version },
+                };
+
+                var identifier = Replace(identifierNode.InnerText, dict);
+
+                dict.Add("DOWNLOADABLE_IDENTIFIER", identifier);
+
+                double.TryParse(fileSizeNode?.InnerText, out var parsedFileSize);
+
+                simulators.Add(new Simulator(
+                    name: Replace(nameNode.InnerText, dict),
+                    identifier: Replace(identifierNode.InnerText, dict),
+                    version: versionNode.InnerText,
+                    source: Replace(sourceNode.InnerText, dict),
+                    installPrefix: Replace(installPrefixNode.InnerText, dict),
+                    fileSize: (long)parsedFileSize));
+            }
+
+            return simulators;
+        }
+
+        protected async Task<Version?> IsInstalled(string identifier)
+        {
+            var (succeeded, pkgInfo) = await ExecuteCommand($"pkgutil", TimeSpan.FromMinutes(1), "--pkg-info", identifier);
+            if (!succeeded)
+            {
+                return null;
+            }
+
+            var lines = pkgInfo.Split('\n');
+            var version = lines.First(v => v.StartsWith("version: ", StringComparison.Ordinal)).Substring("version: ".Length);
+            return Version.Parse(version);
+        }
+
+        private async Task<string?> GetSimulatorIndexXml()
+        {
+            var (xcodeVersion, xcodeUuid) = await GetXcodeInformation();
+
+            var url = $"https://devimages-cdn.apple.com/downloads/xcode/simulators/index-{xcodeVersion}-{xcodeUuid}.dvtdownloadableindex";
```

Comment:
nit: lets move it to a constant. Will the change the url.. I can believe anything,